connectors/sources/github.py

```diff
@@ -1241,10 +1242,60 @@ async def _logged_in_user(self):
         return self._user
 
     async def get_invalid_repos(self):
-        try:
-            self._logger.debug(
-                "Checking if there are any inaccessible repositories configured"
+        self._logger.debug(
+            "Checking if there are any inaccessible repositories configured"
+        )
+        if self.configuration["auth_method"] == GITHUB_APP:
+            return await self._get_invalid_repos_for_github_app()
+        else:
+            return await self._get_invalid_repos_for_personal_access_token()
+
+    async def _get_invalid_repos_for_github_app(self):
+        # A github app can be installed on multiple orgs/personal accounts, so the OWNER in 'OWNER/REPO' is mandatory
+        invalid_repos = set(
+            filter(
+                lambda repo: repo
+                and repo.strip()
+                and len(repo.strip().split("/")) != 2,
+                self.configured_repos,
             )
+        )
+
+        await self._get_installations()
```

Comment:
Because it uses `self._installations` directly. We can either assign it to a local variable `installations`, or we can use `self._installations` directly. 

I choose the latter because it also tries to update `self._installations` (see line 1287), and using `self_installations` makes it clear that we are updating an instance variable (although a local variable will still work)

An alternative is I can change the method name to `_fetch_installations` so that we don't expect it will return anything. WDYT?

Reply:
Might it be better to have method `update_installation` or something and do `installations = await self._get_installations()`?

Reply:
>Might it be better to have method `update_installation` or something

`update_installation` sounds like you can update the installations.

>and do `installations = await self._get_installations()`?

I want to avoid assigning it to a local variable. So all the operations are on `self._installations`.

Let me change the method name to `_fetch_installations`, and add comments